bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/OmnilinkBridgeHandler.java

```diff
@@ -0,0 +1,678 @@
+                    }
+                } else {
+                    logger.warn("Invalid command: {}, must be DecimalType", command);
+                }
+                break;
+            default:
+                logger.warn("Unknown channel for Bridge thing: {}", channelUID.getAsString());
+        }
+    }
+
+    private void makeOmnilinkConnection() {
+        if (omniConnection != null && omniConnection.connected()) {
+            return;
+        }
+
+        logger.debug("Attempting to connect to controller!");
+        try {
+            OmnilinkBridgeConfig config = getThing().getConfiguration().as(OmnilinkBridgeConfig.class);
+
+            this.omniConnection = new Connection(config.getIpAddress(), config.getPort(),
+                    config.getKey1() + ":" + config.getKey2());
+
+            /*
+             * HAI only supports one audio player - cycle through features until we find a feature that is an audio
+             * player.
+             */
+            audioPlayer = reqSystemFeatures().getFeatures().stream()
+                    .map(featureCode -> AudioPlayer.getAudioPlayerForFeatureCode(featureCode))
+                    .filter(Optional::isPresent).findFirst().orElse(Optional.empty());
+
+            systemType = SystemType.getType(reqSystemInformation().getModel());
+
+            if (config.getLogPollingInterval() > 0) {
+                startEventPolling(config.getLogPollingInterval());
+            }
+
+            if (omniConnection != null) {
+                omniConnection.enableNotifications();
+                omniConnection.addNotificationListener(OmnilinkBridgeHandler.this);
+                omniConnection.addDisconnectListener(this);
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            cancelReconnectJob(false);
+            updateBridgeProperties();
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (IOException e) {
+            final Throwable cause = e.getCause();
+            if (cause != null) {
+                final String causeMessage = cause.getMessage();
+
+                if (causeMessage != null && causeMessage.contains("Connection timed out")) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "IP Address probably incorrect, timed out creating connection!");
+                } else if (cause instanceof SocketException) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, causeMessage);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, causeMessage);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        } catch (Exception e) {
```

Comment:
Just a generic exception, it is because of upstream, there is no Specific exception defined. It is thrown when the controller is not accepting new connections

Reply:
If I see correctly the upstream methods are declared with `throws OmniInvalidResponseException, OmniUnknownMessageTypeException, BridgeOfflineException`. You could catch these in a multi-catch.

Reply:
I see, the constructor of  `Connection` in the library is delcared with `throws Exception, IOException, UnknownHostException`. This is a bit suboptimal, but not the scope of this review. You can leave it as is.